frontend/lib/Views/AdminPage/users/edit_user_roles_dialog.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:adpv_frontend/Models/all_users_list_provider.dart';
+import 'package:adpv_frontend/Repository/AdminRepository/users_list_repository.dart';
+import 'package:adpv_frontend/Views/AdminPage/confirmation_dialog_modal.dart';
+import 'package:flutter/material.dart';
+
+import '../groups/group_endpoint_view.dart';
+import '../utils.dart';
+
+const List<String> roles = ["USER", "ADMIN"];
+
+void editUserRoleDialog(
+  BuildContext context,
+  UserListData userListData,
+  AllUsersListProvider provider,
+) {
+  showDialog(
+    context: context,
+    builder: (context) {
+      final List<String> toSelect = List.from(userListData.roles);
+      return StatefulBuilder(
+        builder: (context, setState) => AlertDialog(
+          title: Text(
+            "Choose user roles:",
+            style: defaultAdminTextStyle,
+          ),
+          contentPadding: EdgeInsets.zero,
+          content: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              _buildRolesListTiles(toSelect, setState),
+              Padding(
+                padding: const EdgeInsets.symmetric(vertical: 10),
+                child: _buildButtons(context, userListData, provider, toSelect),
+              )
+            ],
+          ),
+        ),
+      );
+    },
+  );
+}
+
+Column _buildRolesListTiles(List<String> toSelect, StateSetter setState) =>
+    Column(
+      mainAxisSize: MainAxisSize.min,
+      children: roles
+          .map(
+            (element) => ListTile(
+              title: Text(
+                element,
+                style: defaultAdminTextStyle,
+              ),
+              leading: Checkbox(
+                shape: roundedRectangleBorder,
+                activeColor: Colors.teal,
+                value: toSelect.contains(element),
+                onChanged: (bool? value) {
+                  setState(() {
+                    if (value != null && value) {
+                      toSelect.add(element);
+                    } else {
+                      toSelect.remove(element);
+                    }
+                  });
+                },
+              ),
+            ),
+          )
+          .toList(),
+    );
+
+Row _buildButtons(
+  BuildContext context,
+  UserListData userListData,
+  AllUsersListProvider provider,
+  List<String> toSelect,
+) =>
+    Row(
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      children: [
+        _buildCancelButton(context),
+        _buildSaveButton(context, userListData, provider, toSelect)
+      ],
+    );
+
+TextButton _buildCancelButton(BuildContext context) => TextButton(
+      style: ButtonStyle(
+        shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+          RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(5),
+            side: const BorderSide(color: Colors.red),
+          ),
+        ),
+      ),
+      onPressed: () {
+        Navigator.of(context).pop();
+      },
```

Comment:
:thinking: Czy `onPressed` nie powinien być callback'iem? Czy zawsze będziemy chcieli cofać się do tyłu?

Reply:
Task z refactorem ^ mówiłem o tym po kilka razy, komentarz dla statystyk

Reply:
Tutaj nie ma sensu dawać callbacku imo żadnego. Cancel zawsze robi to samo

Reply:
w sumie to @zbrze ma rację